libs/dev-dock/backend/src/backend.ts

```diff
@@ -0,0 +1,176 @@
+// eslint-disable-next-line vx/gts-no-import-export-type
+import type Express from 'express';
+import * as grout from '@votingworks/grout';
+import * as fs from 'fs';
+import { Optional, assert } from '@votingworks/basics';
+import { safeParseElectionDefinition, UserRole } from '@votingworks/types';
+import { isAbsolute, join } from 'path';
+import {
+  CardStatus,
+  readFromMockFile as readFromCardMockFile,
+} from '@votingworks/auth';
+import {
+  isFeatureFlagEnabled,
+  BooleanEnvironmentVariableName,
+} from '@votingworks/utils';
+import { execFile } from './utils';
+
+export type DevDockUserRole = Exclude<UserRole, 'cardless_voter'>;
+export type DevDockUsbDriveStatus = 'inserted' | 'removed';
+export interface DevDockElectionInfo {
+  title: string;
+  path: string;
+}
+
+function electionPathToAbsolute(path: string) {
+  return isAbsolute(path) ? path : join(__dirname, '../../../..', path);
+}
+
+const MOCK_CARD_SCRIPT_PATH = join(
+  __dirname,
+  '../../../auth/scripts/mock-card'
+);
+const MOCK_USB_SCRIPT_DIRECTORY = join(__dirname, '../../../usb-mocking');
+function getUsbDriveStatus(): DevDockUsbDriveStatus {
+  return fs.existsSync('/dev/disk/by-id/usb-mock-part1')
+    ? 'inserted'
+    : 'removed';
+}
+async function removeUsbDrive(): Promise<void> {
+  await execFile('sudo', [join(MOCK_USB_SCRIPT_DIRECTORY, 'remove.sh')]);
+}
+async function insertUsbDrive(): Promise<void> {
+  await execFile('sudo', [join(MOCK_USB_SCRIPT_DIRECTORY, 'insert.sh')]);
+}
+async function clearUsbDrive(): Promise<void> {
+  await execFile('sudo', [
+    join(MOCK_USB_SCRIPT_DIRECTORY, 'initialize.sh'),
+    '-s',
+    '100',
+  ]);
+}
```

Comment:
Most people should have these already enabled but we should call out that people have to edit their `sudoers` file if they haven't already. Maybe you add this later in the PR, haven't seen yet.

Reply:
I see you have this in the `README` already, great!